src/background/Wallet/persistence.ts

```diff
@@ -42,25 +69,130 @@ export class WalletStore extends PersistentStore<WalletStoreState> {
   }
 
   /** Prefer WalletStore['save'] unless necessary */
-  async encryptAndSave(
+  private async encryptAndSave(
     id: string,
-    encryptionKey: string,
+    credentials: Credentials,
```

Comment:
do we need to pass all Credentials here?

Reply:
No, but it makes it more consistent with the other signatures. I think it's more proper to pass credentials, it's easier to abstract them this way